src/Microsoft.DotNet.CodeFormatting.Tests/PropertyBagTests.cs

```diff
@@ -0,0 +1,441 @@
+                PropertyBagValue.SetProperty(TestEnumOptionThree, EmbeddedTestEnumValue);
+                PropertyBagValue.SetProperty(StringSetOption, EmbeddedStringSetValue);
+
+
+                propertyBag.SetProperty(PropertyBagOption, PropertyBagValue);
+
+                TypedPropertyBagValue = propertyBag.GetProperty(TypedPropertyBagOption);
+                TypedPropertyBagValue.SetProperty(TestEnumOptionThree, EmbeddedTestEnumValue);
+
+                propertyBag.SetProperty(TypedPropertyBagOption, TypedPropertyBagValue);
+            }
+
+            internal void ValidatePropertyBag(PropertyBag propertyBag)
+            {
+                Assert.AreEqual(DoubleValue, propertyBag.GetProperty(DoubleOption));
+                Assert.AreEqual(BooleanValue, propertyBag.GetProperty(BooleanOption));
+                Assert.AreEqual(TestEnumValue, propertyBag.GetProperty(TestEnumOptionThree));
+
+                ValidateStringSet(StringSetValue, propertyBag.GetProperty(StringSetOption));
+
+                Assert.AreEqual(EmbeddedDoubleValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(DoubleOption));
+                Assert.AreEqual(EmbeddedBooleanValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(BooleanOption));
+                Assert.AreEqual(EmbeddedTestEnumValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(TestEnumOptionThree));
+
+                ValidateStringSet(EmbeddedStringSetValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(StringSetOption));
+
+                ValidatePropertyBag(PropertyBagValue, propertyBag.GetProperty(PropertyBagOption));
+                ValidatePropertyBag(TypedPropertyBagValue, propertyBag.GetProperty(TypedPropertyBagOption));
+            }
+        }
+
+        private static PerLanguageOption<bool> BooleanOption = 
+            new PerLanguageOption<bool>("TestFeature", "BooleanOption", false);
+
+        private static PerLanguageOption<double> DoubleOption = 
+            new PerLanguageOption<double>("TestFeature", "DoubleOption", 4.7d);
+
+        private static PerLanguageOption<PropertyBag> PropertyBagOption = 
+            new PerLanguageOption<PropertyBag>("TestFeature", "PropertyBagOption", new PropertyBag());
+
+        private static PerLanguageOption<TestEnum> TestEnumOptionTwo =
+            new PerLanguageOption<TestEnum>("TestFeature", "TestEnumOptionTwo", TestEnum.ValueTwo);
+
+        private static PerLanguageOption<TestEnum> TestEnumOptionThree =
+            new PerLanguageOption<TestEnum>("TestFeature", "TestEnumOptionThree", TestEnum.ValueThree);
+
+        private static PerLanguageOption<StringSet> StringSetOption =
+            new PerLanguageOption<StringSet>("TestFeature", "StringSetOption", new StringSet(new string[] { "one", "two" }));
+
+        private static PerLanguageOption<string> StringOption =
+            new PerLanguageOption<string>("TestFeature", "StringOption", null);
+
+        [Serializable]
```

Comment:
Why do you have to explicitly mark an `enum` as `Serializable`? Aren't they always?